src/app/page.tsx

```diff
@@ -1,53 +1,18 @@
 "use client";
-
-import Navbar from "@/components/Navbar";
-import { getPosts } from "../../_actions/postAction";
-import { useInView } from "react-intersection-observer";
-import Image from "next/image";
-import { Button } from "@/components/atoms/button";
-import Link from "next/link";
+import { useRef } from "react";
+import { HeroNavbar } from "@/components/HeroNavbar";
+import { Projects } from "@/components/Projects";
+import { useInView } from "@/hooks/useInView";
 
 export default function Home() {
-  const [ref, inView, entry] = useInView({ threshold: 0.95 });
-
-  // const res = await getPosts();
-  // console.log("Response from getPosts:", res);
+  const contentRef = useRef<HTMLDivElement>(null);
+  const thresholds = Array.from({ length: 100 }, (_, i) => i / 100);
+  const { intersectionRatio } = useInView(contentRef, thresholds);
```

Comment:
What does this do?

Reply:
reference for the custom inview hook to be able to detect the visibility of an element

Reply:
Ah, so gumawa ka ng custom hook using the native intersection observer API